magento/magento2
# Replace Deprecated Guest Shipping Method Management Function

### Description (*)
+ Replaces [deprecated `Magento\Quote\Model\GuestCart\GuestShippingMethodManagement::getShipmentEstimationManagement()` function](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Quote/Model/GuestCart/GuestShippingMethodManagement.php#L106-L118)
    + Removes object manager usage in favour of constructor argument
    + Replaces call to deprecated function with use of class property
    + Removes deprecated `getShipmentEstimationManagement()` function
+ Minor refactoring
    + Simplifies the FQN of the [`$address` parameter of the `estimateByAddress()` function](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Quote/Model/GuestCart/GuestShippingMethodManagement.php#L87)

### Manual testing scenarios (*)
1. Add a product to the cart
2. Proceed to the cart page
3. Estimate shipping using valid address data
4. Ensure the available shipping methods are displayed
5. Ensure the selecting of a shipping method updates cart totals correctly

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#37635: Replace Deprecated Guest Shipping Method Management Function
